### PR TITLE
Ui services departments 152881013

### PR DESCRIPTION
**Pull request by almoral**

### CODE REVIEW

**Comment by almoral** [comment]
@fiallega This has both department and category filtering working.

**Comment by almoral** [comment]
[Finishes #152881182]

File: src/app/shared/services/categories.service.ts [hunk 1]
```diff
@@ -2,34 +2,26 @@ import {Injectable} from '@angular/core';
 import {Observable} from 'rxjs/Observable';
 import {ConfigurationService} from './configuration.service';
 import 'rxjs/Rx';
-import {Category} from '../models/Category';
+import {Filter} from '../models/Filter';
 import * as _ from 'lodash';
 import {HttpClient} from '@angular/common/http';
 
 @Injectable()
 export class CategoriesService {
```
Comment (fiallega) [inline src/app/shared/services/categories.service.ts]: if CategoriesService what it deals with is getting the data from the backend, rename it to CategoryDataService (category-data.service)

File: src/app/shared/services/departments.service.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import {HttpClient} from '@angular/common/http';
+import {ConfigurationService} from './configuration.service';
+import {Filter} from '../models/Filter';
+import {TestDepartments} from '../models/test-departments';
+
+@Injectable()
+export class DepartmentsService {
+
+  constructor( private httpClient: HttpClient,
+               private configurationService: ConfigurationService
```
Comment (fiallega) [inline src/app/shared/services/departments.service.ts]: no need for HttpClient

File: src/app/shared/services/departments.service.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  ) { }
+
+
+  getDepartments(): Observable<Array<Object>> {
+
+    // return this.httpClient.get(this.configurationService.urlDepartments)
+    return Observable.of(TestDepartments.departments)
+      .map((response: any) => {
```
Comment (fiallega) [inline src/app/shared/services/departments.service.ts]: Departments are already set on the data, this.setCalendars(environment.calendars);

File: src/app/shared/services/data-store.service.ts [hunk 4]
```diff
@@ -14,16 +15,19 @@ export class DataStoreService {
...
 
     this.setCalendars(environment.calendars);
 
     this.subscribeTitleFilter();
     this.subscribeCategoriesFilter();
     this.subscribeEvents();
+    this.subscribeCalendarsFilter();
 
     this.getCategories();
+    this.getDepartments();
```
Comment (fiallega) [inline src/app/shared/services/data-store.service.ts]: we are referring to Departments as calendars

**Comment by fiallega** [comment]
Change all Department reference to Calendar

File: src/app/shared/services/data-store.service.ts (hunk 4 above)
Comment (almoral) [reply src/app/shared/services/data-store.service.ts]: You know that was the reason this took me so long right? Calling them calendars is confusing as hell.

File: src/app/shared/models/Filter.ts [hunk 5]
```diff
@@ -3,7 +3,7 @@
  */
```
Comment (fiallega) [inline src/app/shared/models/Filter.ts]: Filter.js should be called filter.js

File: src/app/shared/services/departments.service.ts (hunk 2 above)
Comment (almoral) [reply src/app/shared/services/departments.service.ts]: We're never going to get them from an endpoint?

File: src/app/shared/services/departments.service.ts (hunk 3 above)
Comment (almoral) [reply src/app/shared/services/departments.service.ts]: I thought you asked me to hardcode them?

**Comment by fiallega** [comment]
I dont like the idea of combining "Filter" as categories and calendars (departments)

File: src/app/shared/models/Filter.ts (hunk 5 above)
Comment (almoral) [reply src/app/shared/models/Filter.ts]: No problem.

**Comment by almoral** [comment]
@fiallega I made the changes you requested so it should be good to go.

File: src/app/shared/models/test-departments.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import {environment} from '../../../environments/environment';
+
+export class TestDepartments {
+
```
Comment (fiallega) [inline src/app/shared/models/test-departments.ts]: if it is not getting used, delete it (TestDepartments) I could not find any reference,

File: src/app/shared/models/calendar.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+
+  /**
+   * fromJson creates an instance of the MDCEvent object from
+   * a json. If the json does not validate correctly, null is returned.
+   * @param json - designed to be used with a json coming from a service.
+   * @returns {any} if json validates returns an MDCEvent otherwise it returns null.
+   */
+  public static fromJSON(json: any): Calendar {
+    // if (MDCEvent.validateJson(json))
```
Comment (fiallega) [inline src/app/shared/models/calendar.ts]: is not coming from json .. right ?

**Comment by fiallega** [comment]
I am confused on how do you create Calendars

File: src/app/shared/services/data-store.service.ts [hunk 8]
```diff
@@ -14,16 +15,19 @@ export class DataStoreService {
 
   constructor(private eventService: EventService,
               private eventDataService: EventDataService,
-              private categoriesService: CategoriesService) {
+              private categoriesService: CategoriesDataService,
+              private departmentsService: CalendarDataService) {
```
Comment (fiallega) [inline src/app/shared/services/data-store.service.ts]: call it calendarDataService if in deed this service is needed

File: src/app/shared/models/calendar.ts (hunk 7 above)
Comment (almoral) [reply src/app/shared/models/calendar.ts]: You are correct. I will get rid of that.

File: src/app/shared/services/data-store.service.ts (hunk 8 above)
Comment (almoral) [reply src/app/shared/services/data-store.service.ts]: Oh wow, I missed that one.

File: src/app/shared/services/calendar-data.service.ts [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+   * maps the raw array of json topics into an array
+   * of Filter Objects. If any of the elements in the
+   * array can not be mapped, it is ignored.
+   * @param response - Response object containing a body equal to json array representing raw topics
+   * @returns {Array<Calendar>} Converted response into an array
+   * of Filter Objects. It will return an empty array if nothing
+   * in the response could be converted to an Topic or the response hand an empty body.
+   */
+  private jsonToCalendars = (response: any) => {
+    if (_.isEmpty(response)) {
```
Comment (fiallega) [inline src/app/shared/services/calendar-data.service.ts]: nobody is using jsonToCalendars. It is commented out above, therefore what is the need of Calendar.ts

Comment (almoral) [reply src/app/shared/services/calendar-data.service.ts]: @fiallega It's been removed

**Pull request merged**